ckanext/dcatapchharvest/tests/test_helpers.py

```diff
@@ -0,0 +1,172 @@
+                {
+                    "modified": "2020-01-02T00:00:00",
+                },
+            ],
+        }
+        dataset_dict = {
+            "resources": [
+                {
+                    "modified": "2020-01-02T12:00:00",
+                },
+            ],
+        }
+
+        assert check_package_change(existing_package, dataset_dict) == (
+            True,
+            "resource modified date changed: 2020-01-02T12:00:00"
+        )
+
+    def test_check_package_change_new_resource_url(self):
+        existing_package = {
+            "resources": [
+                {
+                    "url": "http://example.org/resource-1",
+                },
+            ],
+        }
+        dataset_dict = {
+            "resources": [
+                {
+                    "url": "http://example.org/new/resource-1",
+                },
+            ],
+        }
+
+        assert check_package_change(existing_package, dataset_dict) == (
+            True,
+            "resource access url changed: http://example.org/new/resource-1"
+        )
+
+    def test_check_package_change_new_resource_download_url(self):
+        existing_package = {
+            "resources": [
+                {
+                    "download_url": "http://example.org/resource-1/download",
+                },
+            ],
+        }
+        dataset_dict = {
+            "resources": [
+                {
+                    "download_url": "http://example.org/new/resource-1/download",
+                },
+            ],
+        }
+
+        assert check_package_change(existing_package, dataset_dict) == (
+            True,
+            "resource download url changed: http://example.org/new/resource-1/download"
+        )
+
+    def test_check_package_change_different_resource_count(self):
+        existing_package = {
+            "resources": [
+                {
+                    "url": "http://example.org/resource-1",
+                    "download_url": "http://example.org/resource-1/download",
+                    "modified": "2020-01-02T00:00:00",
+                },
+                {
+                    "url": "http://example.org/resource-2",
+                    "download_url": "http://example.org/resource-2/download",
+                    "modified": "2020-01-02T00:00:00",
+                },
+            ],
+        }
+        dataset_dict = {
+            "resources": [
+                {
+                    "url": "http://example.org/resource-1",
+                    "download_url": "http://example.org/resource-1/download",
+                    "modified": "2020-01-02T00:00:00",
+                },
+            ],
+        }
+
+        assert check_package_change(existing_package, dataset_dict) == (
+            True,
+            "resource count changed: 1"
+        )
+
+    def test_check_package_change_multiple_resources_changed(self):
+        """The number of resources have changed and so have all the urls,
```

Comment:
this description I do not understand, I see the same # of resources in existing_package and dataset_dict